src/actions/Highlight.ts

```diff
@@ -10,12 +15,33 @@ export default class Highlight implements Action {
 
   async run(
     [targets]: [Target[]],
-    styleName: EditStyleName = "highlight0",
+    highlightId?: HighlightId,
   ): Promise<ActionReturnValue> {
-    const style = this.graph.editStyles[styleName];
+    if (ide().capabilities.commands["highlight"] == null) {
+      throw Error(`The highlight action is not supported by your ide`);
+    }
 
-    this.graph.editStyles.clearDecorations(style);
-    await this.graph.editStyles.setDecorations(targets, style);
+    if (targets.length === 0) {
+      // Special case to clear highlights for the active editor when user says
+      // "highlight nothing"
+      const { activeTextEditor } = ide();
+
+      if (activeTextEditor == null) {
+        throw Error(
+          "The `highlight nothing` command requires an active text editor",
+        );
+      }
+
+      await ide().setHighlightRanges(highlightId, activeTextEditor, []);
+    } else {
```

Comment:
I just ended up putting a special case in here, because otherwise we'd break `"highlight nothing"` if we didn't have a migration, which doesn't seem worth the effort for something that is confined to an action that I'm guessing is very uncommonly used